src/components/organisms/BannerAdmin/BannerAdmin.tsx

```diff
@@ -1,90 +1,176 @@
-import React, { useState, useCallback, useRef } from "react";
+import React, { useCallback, useState, useEffect } from "react";
+import { useForm } from "react-hook-form";
 
 import { makeUpdateBanner } from "api/bannerAdmin";
 
 import { AnyVenue } from "types/venues";
+import { Banner } from "types/banner";
+
+import { useShowHide } from "hooks/useShowHide";
+
+import { Button } from "components/atoms/Button";
+import { Checkbox } from "components/atoms/Checkbox";
+import { InputField } from "components/atoms/InputField";
+
+import { ConfirmationBannerModal } from "./ConfirmationBannerModal";
+
+import "./BannerAdmin.scss";
 
 interface BannerAdminProps {
   venueId?: string;
   venue: AnyVenue;
+  onClose: () => void;
 }
 
 // @debt This component is almost exactly the same as IframeAdmin, we should refactor them both to use the same generic base component
 //   BannerAdmin is the 'canonical example' to follow when we do this
-export const BannerAdmin: React.FC<BannerAdminProps> = ({ venueId, venue }) => {
-  const existingBannerMessage = venue?.bannerMessage ?? "";
-
-  const inputFieldRef = useRef<HTMLInputElement>(null);
-  const [error, setError] = useState<string | null>();
+export const BannerAdmin: React.FC<BannerAdminProps> = ({
+  venueId,
+  venue,
+  onClose,
+}) => {
+  const { register, handleSubmit, errors, reset, watch } = useForm<Banner>({
+    mode: "onChange",
+    reValidateMode: "onChange",
+  });
+  const isActionButtonValue = watch("isActionButton");
+
+  const [bannerData, setBannerDate] = useState<Banner>();
+
+  const {
+    isShown: isShowModal,
+    show: showModal,
+    hide: hideModal,
+  } = useShowHide();
+  const [isDisabledUrlFields, setIsDisabledUrlFields] = useState(
+    venue?.banner?.isActionButton
+  );
 
-  const handleInputChange = useCallback(() => {
-    setError(null);
-  }, []);
+  useEffect(() => {
+    setIsDisabledUrlFields(isActionButtonValue);
+  }, [isActionButtonValue]);
 
   const updateBannerInFirestore = useCallback(
-    (msg?: string) => {
+    (banner?: Banner) => {
       if (!venueId) return;
-
-      makeUpdateBanner(venueId, (errorMsg) => setError(errorMsg))(msg);
+      makeUpdateBanner({ venueId, banner });
+      onClose();
```

Comment:
If you follow my comments above to make this optional then you will need to do something like this here:
```suggestion
      onClose && onClose();
```

Reply:
would it be better to introduce a default value for the onClose, say `() => {}` or a check like you've proposed `onClose && onClose()` . Asking to see if one of those is better than the other?

Reply:
Generally I tend to prefer not using ‘arbitrary’ defaults when we can ‘tell the truth’ about it being empty by leaving it undefined. So I would likely leave it undefined and do the check as described here

Reply:
changed in 771b70a

Reply:
fixed 1c2d835